database_pregel.go

```diff
@@ -0,0 +1,182 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+
+package driver
+
+import (
+	"context"
+	"time"
+)
+
+// DatabasePregels provides access to all Pregel Jobs in a single database.
+type DatabasePregels interface {
+	// StartJob - Start the execution of a Pregel algorithm
+	StartJob(ctx context.Context, options PregelJobOptions) (string, error)
+	// GetJob - Get the status of a Pregel execution
+	GetJob(ctx context.Context, id string) (*PregelJob, error)
+	// GetJobs - Returns a list of currently running and recently finished Pregel jobs without retrieving their results.
+	GetJobs(ctx context.Context) ([]*PregelJob, error)
+	// CancelJob - Cancel an ongoing Pregel execution
+	CancelJob(ctx context.Context, id string) error
+}
+
+type PregelAlgorithm string
+
+const (
+	PregelAlgorithmPageRank                        PregelAlgorithm = "pagerank"
+	PregelAlgorithmSingleSourceShortestPath        PregelAlgorithm = "sssp"
+	PregelAlgorithmConnectedComponents             PregelAlgorithm = "connectedcomponents"
+	PregelAlgorithmWeaklyConnectedComponents       PregelAlgorithm = "wcc"
+	PregelAlgorithmStronglyConnectedComponents     PregelAlgorithm = "scc"
+	PregelAlgorithmHyperlinkInducedTopicSearch     PregelAlgorithm = "hits"
+	PregelAlgorithmEffectiveCloseness              PregelAlgorithm = "effectivecloseness"
+	PregelAlgorithmLineRank                        PregelAlgorithm = "linerank"
+	PregelAlgorithmLabelPropagation                PregelAlgorithm = "labelpropagation"
+	PregelAlgorithmSpeakerListenerLabelPropagation PregelAlgorithm = "slpa"
+)
+
+type PregelJobOptions struct {
+	// Name of the algorithm
+	Algorithm PregelAlgorithm `json:"algorithm"`
+	// Name of a graph. Either this or the parameters VertexCollections and EdgeCollections are required.
+	// Please note that there are special sharding requirements for graphs in order to be used with Pregel.
+	GraphName string `json:"graphName,optional"`
+	// List of vertex collection names. Please note that there are special sharding requirements for collections in order to be used with Pregel.
+	VertexCollections []string `json:"vertexCollections,optional"`
+	// List of edge collection names. Please note that there are special sharding requirements for collections in order to be used with Pregel.
+	EdgeCollections []string `json:"edgeCollections,optional"`
+	// General as well as algorithm-specific options.
```

Comment:
```suggestion
	// General and algorithm-specific options.
```

Reply:
this field is general as general are alg-spec options..